packages/hardhat-core/src/internal/hardhat-network/provider/input.ts

```diff
@@ -28,6 +28,16 @@ const isRpcDataString = (u: unknown) =>
 const isRpcHashString = (u: unknown) =>
   typeof u === "string" && u.length === 66 && isRpcDataString(u);
 
+// A bug in isValidAddress makes it throw
+// if the string isn't hex prefixed...
+const safeIsValidAddress = (u: unknown) => {
+  let isValid = false;
+  try {
+    isValid = isValidAddress(u as string);
+  } catch (e) {}
+  return isValid;
+};
+
```

Comment:
On reflection I don't think this is a bug...it's just input validation that's inconvenient here.

Reply:
Ok, you're right.